libres/lib/enkf/enkf_main.cpp

```diff
@@ -519,6 +519,13 @@ int enkf_main_load_from_run_context(enkf_main_type *enkf_main,
     Semafoor concurrently_executing_threads(100);
     std::vector<std::tuple<int, std::future<int>>> futures;
 
+    // If this function is called via pybind11 we need to release
+    // the GIL here because this function may spin up several
+    // threads which also may need the GIL (e.g. for logging)
+    PyThreadState *state = nullptr;
```

Comment:
Using the pybind11 `py::scoped_gil_release` here should be safer than using the C API due to RAII (`logger->warning` may throw an exception via Python, for example).

Reply:
My first thought also, but it blows up due to some tests calling these functions directly from C++ (i.e. not holding the GIL, then trying to release it). Moreover, the `scoped_gil_release` is (as name indicates) indeed scoped, so it cannot be used conditionally with an if-check like the current patch uses.

Reply:
@pinkwah further comments?